### PR TITLE
NO-ISSUE: use appropriate image in operator upgrade

### PR DESCRIPTION
**Pull request by danielerez**
# Assisted Pull Request

## Description
The operator upgrade flow requires using the previous OCP version if ASSISTED_UPGRADE_OPERATOR is true. Otherwise, use the latest version. Hence, to ensure we fetch the proper version, sorted the release images list and filtered out non-default cpu architecture (as not relevant for the operator).

E.g.
* When ASSISTED_UPGRADE_OPERATOR is false:
ASSISTED_OPENSHIFT_VERSION = openshift-v4.9
ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE = quay.io/openshift-release-dev/ocp-release:4.9.0-x86_64
* When ASSISTED_UPGRADE_OPERATOR is true:
ASSISTED_OPENSHIFT_VERSION = openshift-v4.8
ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE = quay.io/openshift-release-dev/ocp-release:4.8.13-x86_64

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [x] assisted-test-infra environment
- [ ] dev-scripts environment
- [x] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @pawanpinjarkar 
/cc @osherdp 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by osherdp — APPROVED** [review]
Looks good to me
But is there no test we can run to validate it?
/hold

**Comment by pawanpinjarkar** [comment]
Tested locally and /lgtm

File: deploy/operator/common.sh [hunk 1]
```diff
@@ -26,15 +26,20 @@ export ASSISTED_SERVICE_OPERATOR_CATALOG="assisted-service-operator-catalog"
...
+    RELEASE_IMAGE=$(echo ${SORTED_RELEASE_IMAGES} | jq '[.[].url][-1]')
+    VERSION=$(echo ${SORTED_RELEASE_IMAGES} | jq '[.[].openshift_version][-1]')
 else
     # Before the AI operator upgrade, we install the version prior to the most current one of OCP. 
-    # The most current version of OCP we are installing is 4.9, and the version previous to that is 4.8.
-    export RELEASE_IMAGE=$(echo ${DEFAULT_OS_IMAGES} | jq -rc '[.[].url][-2]')
-    export VERSION=$(echo ${DEFAULT_OS_IMAGES} | jq -rc '[.[].openshift_version][-2]')
+    # E.g. the most current version of OCP we are installing is 4.9, and the version previous to that is 4.8.
+    RELEASE_IMAGE=$(echo ${SORTED_RELEASE_IMAGES} | jq '[.[].url][-2]')
+    VERSION=$(echo ${SORTED_RELEASE_IMAGES} | jq '[.[].openshift_version][-2]')
```
Comment (pawanpinjarkar) [inline deploy/operator/common.sh:42]: extra `"` in VERSION var fails the ClusterImageSet name validation 
```
kind: ClusterImageSet
metadata:
  name: 'openshift-v"4.9"'
```
[here](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_assisted-service/2808/pull-ci-openshift-assisted-service-master-e2e-metal-assisted-operator-ztp/1451155747000815616#1:build-log.txt%3A572)

File: deploy/operator/common.sh [hunk 2]
```diff
@@ -26,15 +26,20 @@ export ASSISTED_SERVICE_OPERATOR_CATALOG="assisted-service-operator-catalog"
...
 fi
 
 export ASSISTED_OPENSHIFT_VERSION="${ASSISTED_OPENSHIFT_VERSION:-openshift-v${VERSION}}"
```
[the remark this answers is not in the document]
Comment (danielerez) [reply deploy/operator/common.sh:45]: done

**Pull request merged**